lib/instrumentation/noop-transaction.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+const constants = require('../constants');
+
+module.exports = NoopTransaction;
+
+const NOOP_TRANSACTION_ID = '0000000000000000';
+const NOOP_TRACEID = '00000000000000000000000000000000';
+const NOOP_TRACEPARENT = '00-00000000000000000000000000000-0000000000000000-00';
+
+// Usage:
+//    new NoopTransaction()
+function NoopTransaction() {
+  // Public properties:
+  // https://www.elastic.co/guide/en/apm/agent/nodejs/current/transaction-api.html
+  this.name = 'unnamed';
+  this.type = 'noop';
+  this.subtype = undefined;
+  this.action = undefined;
+  this.traceparent = NOOP_TRACEPARENT;
+  this.result = constants.RESULT_SUCCESS;
+  this.outcome = 'unknown';
+  this.ids = {
+    'trace.id': NOOP_TRACEID,
+    'transaction.id': NOOP_TRANSACTION_ID,
+  };
+
+  // Non official mentioned in index.d.ts
+  this.timestamp = Date.now();
+  this.id = NOOP_TRANSACTION_ID;
+  this.traceId = NOOP_TRACEID;
+  this.sampled = false;
+  this.ended = false;
+}
+
+// Public methods:
+// https://www.elastic.co/guide/en/apm/agent/nodejs/current/transaction-api.html
+NoopTransaction.prototype.setType = returnVoid;
+NoopTransaction.prototype.setLabel = returnTrue;
+NoopTransaction.prototype.addLabels = returnTrue;
+NoopTransaction.prototype.setOutcome = returnVoid;
+NoopTransaction.prototype.startSpan = returnNull;
+NoopTransaction.prototype.end = returnVoid;
+NoopTransaction.prototype.ensureParentId = function () {
+  return '00000000';
```

Comment:
```
> var t = apm.startTransaction('t1')
undefined
> t.ensureParentId()
'dccd55f81e5a251c'
```

I think the retval of `ensureParentId` needs to be 8-bytes (16 hex chars).
It could re-use `NOOP_TRANSACTION_ID`.

```suggestion
  return NOOP_TRANSACTION_ID;
```